### PR TITLE
add column level ACL

### PR DESCRIPTION
**Pull request by vijayrathore8492**
## Change Summary

Added visibility_rules in the column. eg. 
`[{"access": "deny", "roles":["editor"]}]` will deny access of the column from all editors.
`[{"access": "allow", "roles":["creator"]}]` will deny access to all roles accept creator.

## Change type

- [ ] feat: (new feature for the user, not a new feature for build script)
- [ ] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi-colons, etc; no production code change)
- [ ] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

Provide summary of changes.

## Additional information / screenshots (optional)

Anything for maintainers to be made aware of

### CODE REVIEW

File: packages/nocodb/src/lib/db/sql-data-mapper/lib/sql/BaseModelSqlv2.ts [hunk 1]
```diff
@@ -1682,6 +1723,25 @@ class BaseModelSqlv2 {
     }
   }
 
+  private async mapAliasToColumn(data) {
```
Comment (vijayrathore8492) [inline packages/nocodb/src/lib/db/sql-data-mapper/lib/sql/BaseModelSqlv2.ts:1726]: copied this function from `Model.ts` to here.

File: packages/nocodb/src/lib/meta/api/dataApis/bulkDataAliasApis.ts [hunk 2]
```diff
@@ -15,58 +15,63 @@ async function bulkDataInsert(req: Request, res: Response) {
...
     viewId: view?.id,
     dbDriver: NcConnectionMgrv2.get(base),
+    userRoles: req?.session?.passport?.user?.roles,
   });
 
   res.json(await baseModel.bulkInsert(req.body, { cookie: req }));
 }
 
-async function bulkDataUpdate(req: Request, res: Response) {
+async function bulkDataUpdate(req, res: Response) {
```
Comment (vijayrathore8492) [inline packages/nocodb/src/lib/meta/api/dataApis/bulkDataAliasApis.ts:24]: removed type so it does not show an error when accessing `req?.session`

File: packages/nocodb-sdk/src/lib/Api.ts [hunk 3]
```diff
@@ -283,6 +284,12 @@ export interface LinkToAnotherRecordType {
   order?: string;
 }
 
+export interface ColumnVisibilityRuleType {
+  effect?: 'deny' | 'read' | 'write';
```
Comment (ketewan) [inline packages/nocodb-sdk/src/lib/Api.ts:288]: Wouldn't "access" or "permissions" be better? To be more consistent with commonly used terms.
Where does "read and write" permission fit in?

Comment (vijayrathore8492) [reply packages/nocodb-sdk/src/lib/Api.ts:288]: For now, read and write both works as same. I will change the names for better understanding.

**Review by ketewan — APPROVED** [review]

**Pull request merged**